packages/@aws-cdk/aws-ec2/lib/client-vpn-endpoint-types.ts

```diff
@@ -0,0 +1,57 @@
+import { IDependable, IResource } from '@aws-cdk/core';
+import { IConnectable } from './connections';
+
+/**
+ * A client VPN endpoint
+ */
+export interface IClientVpnEndpoint extends IResource, IConnectable {
+  /**
+   * The endpoint ID
+   */
+  readonly endpointId: string;
+
+  /**
+   * Dependable that can be depended upon to force target networks associations
+   */
+  readonly targetNetworksAssociated: IDependable;
+}
+
+/**
+ * A Lambda function
+ */
+export interface IFunction {
+  /**
+   * The ARN of the function.
+   */
+  readonly functionArn: string;
+}
+
+/**
+ * A certificate
+ */
+export interface ICertificate {
```

Comment:
Some for this, although the dependency is trickier. We don't have a dependency yet and adding one feels unsafe. So definitely looks like it needs an integration package.

Reply:
Do we need a `ICertifcate` here after all? Would it be here acceptable to reference an `ARN` (string) directly in `ClientVpnEndpointOptions`? I mean how likely is it that CF adds support for "imported" certificates in ACM?

Reply:
Not sure what "imported" certificates means, in this case.

I guess just taking an ARN is acceptable for now... I suppose we can deprecate that field once we figure out a good way to do the integration. Everything I can think of is pretty bad, so just taking a string seems like the simplest solution.

Reply:
Imported means this: https://docs.aws.amazon.com/acm/latest/userguide/import-certificate.html

Certificates automatically issued by ACM cannot be used here for the client VPN endpoint.

Changed for now to `serverCertificateArn` and `clientCertificateArn` but another option could be to define `IClientVpnCertificate` and when CF adds a resource in ACM for imported certificates it will `implement` it?

I see that CF support for [`AWS::IAM::ServerCertificate`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-servercertificate.html), what we need here is the same but in ACM.